# HG changeset patch
# User Rik <rik@octave.org>
# Date 1360286470 28800
#      Thu Feb 07 17:21:10 2013 -0800
# Node ID 9194bb707d63e904596a5a24b4c35f601102e45e
# Parent  d910b9211cca45f00a19e3134c3070e45a63bfa5
Plug memory leak in edit_history.

* libinterp/interpfcn/oct-hist.cc(do_edit_history): delete string
returned from edit_history_readline.

diff --git a/libinterp/interpfcn/oct-hist.cc b/libinterp/interpfcn/oct-hist.cc
--- a/libinterp/interpfcn/oct-hist.cc
+++ b/libinterp/interpfcn/oct-hist.cc
@@ -527,16 +527,18 @@ do_edit_history (const octave_value_list
 
       if (first)
         {
           first = 0;
           edit_history_repl_hist (line);
         }
       else
         edit_history_add_hist (line);
+
+      delete [] line;
     }
 
   file.close ();
 
   // Turn on command echo, so the output from this will make better
   // sense.
 
   unwind_protect frame;
